From 42e1c272734ff3880fe3e01bd8a967fb9efc1e0e Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 18 Mar 2020 13:26:37 +0200
Subject: [PATCH] [SW-10047] goya2_arc: pll: remove all config data

The PLLs configuration handling is moving to U-Boot in order to minimize
the preboot image.
Remove the PLLs handling from preboot.

Change-Id: I211477745fe9ce7d8da688fd6f345ced2778bbd3
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/goya2_arc/asic.c b/goya2_arc/asic.c
index 5fb7a50..c7ce08e 100644
--- a/goya2_arc/asic.c
+++ b/goya2_arc/asic.c
@@ -5,8 +5,6 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
-#include <stdbool.h>
-
 #include <psoc_global_conf_regs.h>
 
 #include <preboot.h>
@@ -14,64 +12,6 @@
 #include <uart.h>
 
 #include "common.h"
-
-static struct hl_pll_cfg goya_pll_cfg[] = {
-	{
-		.name = "IC", .base = mmIC_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 15, .nb = 31,
-		.div_factor = { 0, 1, 1, 2 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "EMMC", .base = mmPSOC_EMMC_PLL_BASE,
-		.nr = 24, .nf = 415, .od = 15, .nb = 207,
-		.div_factor = { 0, 1, -1, -1 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
-		},
-	}, {
-		.name = "MME", .base = mmPSOC_MME_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 15, .nb = 31,
-		.div_factor = { 0, 1, 2, 8 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "PCI", .base = mmPSOC_PCI_PLL_BASE,
-		.nr = 4, .nf = 343, .od = 3, .nb = 171,
-		.div_factor = { 0, 35, 8, 85 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "TPC", .base = mmTPC_PLL_BASE,
-		.nr = 0, .nf = 63, .od = 15, .nb = 31,
-		.div_factor = { 0, 1, 1, 8 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	},
-};
-
-int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
-{
-	int pll_num = ARRAY_SIZE(goya_pll_cfg);
-
-	*hl_pll_cfg = goya_pll_cfg;
-
-	return pll_num;
-}
 
 u32 hl_skip_pcie_init(void)
 {
@@ -102,4 +42,3 @@
 
 	return factor > 2 ? factor : 2;
 }
-
